GiEnJul/Controllers/AdminController.cs

```diff
@@ -132,6 +132,56 @@ public async Task<ActionResult> PutWishAsync(string person_rowkey, [FromBody] st
             return Ok();
         }
 
+        [HttpDelete("Connection")]
+        [Authorize(Policy = "DeleteConnection")]
+        public async Task<ActionResult> DeleteConnectionAsync(string location, string rowKey)
+        {
+            var giver = await _giverRepository.GetGiverAsync(location, rowKey);
+
+            if (giver?.MatchedRecipient is null)
+            {
+                return NotFound();
+            }
+
+            var recipient = await _recipientRepository.GetRecipientAsync(location, giver.MatchedRecipient);
+
+            if (recipient is null)
+            {
+                return NotFound();
+            }
+
+            var originalGiver = giver.ShallowCopy();
+            var originalRecipient = recipient.ShallowCopy();
+
+            giver.HasConfirmedMatch = false;
+            giver.IsSuggestedMatch = false;
+            giver.MatchedRecipient = null;
+
+            recipient.HasConfirmedMatch = false;
+            recipient.IsSuggestedMatch = false;
+            recipient.MatchedGiver = null;
+
+            try
+            {
+                await _giverRepository.InsertOrReplaceAsync(giver);
+                await _recipientRepository.InsertOrReplaceAsync(recipient);
+
+                if (await _connectionRepository.ConnectionExists(giver, recipient))
```

Comment:
This if and contents should be put outside the try catch. Should the if be false, the giver and recipient while no connection is updated.

The if and its contents should probably be put before the try catch.